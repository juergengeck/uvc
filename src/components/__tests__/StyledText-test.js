import * as React from 'react';
import renderer from 'react-test-renderer';
import { MonoText } from '../StyledText';

describe('StyledText', () => {
  it('renders MonoText with SpaceMono font family', () => {
    const tree = renderer.create(<MonoText>Test text</MonoText>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.props.style).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          fontFamily: 'SpaceMono'
        })
      ])
    );
  });

  it('preserves additional styles passed as props', () => {
    const customStyle = { color: 'red', fontSize: 16 };
    const tree = renderer.create(<MonoText style={customStyle}>Test text</MonoText>).toJSON();
    expect(tree.props.style).toEqual(
      expect.arrayContaining([
        customStyle,
        expect.objectContaining({
          fontFamily: 'SpaceMono'
        })
      ])
    );
  });
});
