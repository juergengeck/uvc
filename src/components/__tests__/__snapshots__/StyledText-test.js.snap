// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText renders MonoText with SpaceMono font family 1`] = `
<Text
  style={
    [
      undefined,
      {
        "fontFamily": "SpaceMono",
      },
    ]
  }
>
  Test text
</Text>
`;
